### DISCUSSION TITLE
SExtractor build for Windows

### DISCUSSION THREAD

**Discussion by blackhaz** [astromatic/sextractor#53] [General]
Dear All,

Has anybody compiled SExtractor for Windows? Would appreciate if someone could post a link to EXE/archive with all the files to avoid building it myself.

Thanks!

**Comment by PhilipBTaylor** [comment]
A couple of years ago we investigated a Windows build of SExtractor and saw that it would be very difficult to do a native build on Windows. Instead, we cross-compiled on Linux for a Windows target. See the Release files (which include a .exe file) at
https://github.com/Observatory-Sciences/sextractor

**Comment by jmgoldba** [comment]
PBT, you're a better man than I! I just finished building SExtractor on Cygwin and boy, it wasn't easy. I'd not gotten around to testing it yet when I found your post so I've got two to try. After some trial and error, the TL;DR version of my effort was to install Cygwin and the basic build packages (e.g. make, gcc, &c.) using the Cygwin setup that you download. Then you: $ ./configure --disable-model-fitting

The configure utility will let you know if you missed a package, e.g. one I missed was CFITSIO. I stumbled on the disable-model-fitting while googling issues I was originally having trying to build/install ATLAS(?) from source.

Before the long story, I'll note I've been using a Windows version of SExtractor that comes with Bob Denny's PinPoint. It was built with Visual Studio and works fairly well, but it's an older version that was missing a feature I decided I wanted to try. Bob wasn't interested in updating it and VS is, well. VS. 

Long story. Alas no model fitting because, damn, ATLAS was too much for little me to install (correctly) from behind, viz. scratch. While I (and ./configure) had no issues with Cygwin FFTW, giving Cygwin's openBLAS a go was a no go. Mostly because there's no -dev version to install. (You need dev versions if you need to develop, viz. include and link to stuff.) I found a site somewhere that said no problem, install lapack-dev and point your shizzle at that but no:

./configure --enable-openblas --with-openblas-incdir=/usr/include --with-openblas-libdir=/usr/lib

came up with:

checking for /usr/include/cblas.h... yes
checking for library containing LAPACKE_dpotrf... no
configure: error: OpenBLAS library file not found! Exiting.

I couldn't find any LAPACKE routines after the lapack install and decided [no mas](https://en.wikipedia.org/wiki/Roberto_Dur%C3%A1n_vs._Sugar_Ray_Leonard_II)!

Jesse
